authorization/invitation/service/invitation_service.go

```diff
@@ -240,10 +240,14 @@ func (s *invitationServiceImpl) processTeamInviteNotifications(ctx context.Conte
 	teamName := team.IdentityResource.Name
 
 	var spaceName string
+	res, err := s.Repositories().ResourceRepository().Load(ctx, team.IdentityResourceID.String)
+	if err != nil {
+		return err
+	}
 
 	// Every team *should* have a parent space, but we'll put this check here just in case
-	if team.IdentityResource.ParentResourceID != nil {
-		sp, err := s.Services().WITService().GetSpace(ctx, *team.IdentityResource.ParentResourceID)
+	if res.ParentResourceID != nil {
+		sp, err := s.Services().WITService().GetSpace(ctx, *res.ParentResourceID)
```

Comment:
you could use the `CheckExists()` method if you just want to make sure that the parent space exists.

Reply:
also, what would happen if the parent space did not exist ?

Reply:
I think if that's the case we are sending empty space name in the notification message

Reply:
AFAICT, you have not used the `CheckExists` method. Also, if the space name is empty, should there be a notification sent to the user?